code/datums/components/growth_and_differentiation.dm

```diff
@@ -44,6 +52,11 @@
 	src.optional_checks = optional_checks
 	src.optional_grow_behavior = optional_grow_behavior
 
+	if(islist(signals_to_kill_on))
+		src.signals_to_kill_on = signals_to_kill_on
+		for(var/signal in src.signals_to_kill_on)
+			RegisterSignal(parent, signal, PROC_REF(kill_processes))
+
```

Comment:
Should go into `RegisterWithParent`? Doesn't matter since the comp is not transferable but you're using `UnregisterFromParent` so, consistency